yt/yt/tests/integration/queues/test_queue_agent.py

```diff
@@ -3995,6 +4013,111 @@ def assert_exported_table_count(count):
 
         self.remove_export_destination(export_dir)
 
+    @authors("pavel-bash")
+    def test_cron_annotation_schedule(self):
+        # Check that the new CRON annotation works correctly, at least for this use-case.
+        if getattr(self, "USE_OLD_QUEUE_EXPORTER_IMPL"):
+            pytest.skip()
+
+        _, queue_id = self._create_queue("//tmp/q")
+
+        export_dir = "//tmp/export"
+        self._create_export_destination(export_dir, queue_id)
+
+        export_cron_expression = "0/5 * * * * *"  # Every 5 seconds.
+
+        set("//tmp/q/@static_export_config", {
+            "default": {
+                "export_directory": export_dir,
+                "export_cron_schedule": export_cron_expression,
+            }
+        })
+
+        self._wait_for_component_passes()
+
+        insert_rows("//tmp/q", [{"data": "vim"}])
+        self._flush_table("//tmp/q")
+        wait(lambda: len(ls(export_dir)) == 1, timeout=7)
+
+        insert_rows("//tmp/q", [{"data": "nano"}])
+        self._flush_table("//tmp/q")
+        wait(lambda: len(ls(export_dir)) == 2, timeout=7)
+
+        self.remove_export_destination(export_dir)
+
+    @authors("pavel-bash")
+    def test_cron_aperiodic_interval(self):
+        if getattr(self, "USE_OLD_QUEUE_EXPORTER_IMPL"):
+            pytest.skip()
+
+        _, queue_id = self._create_queue("//tmp/q")
+
+        export_dir = "//tmp/export"
+        self._create_export_destination(export_dir, queue_id)
+
+        export_seconds = [0, 17, 29, 45]
+        export_cron_expression = "0,17,29,45 * * * * *"  # Every ~15 seconds.
+
+        set("//tmp/q/@static_export_config", {
+            "default": {
+                "export_directory": export_dir,
+                "export_cron_schedule": export_cron_expression,
+            }
+        })
+
+        self._wait_for_component_passes()
+
+        insert_rows("//tmp/q", [{"data": "vim"}])
+        self._flush_table("//tmp/q")
+        wait(lambda: len(ls(export_dir)) == 1, timeout=20)
+        time1_second = datetime.datetime.now().second
+
+        insert_rows("//tmp/q", [{"data": "nano"}])
+        self._flush_table("//tmp/q")
+        wait(lambda: len(ls(export_dir)) == 2, timeout=20)
+        time2_second = datetime.datetime.now().second
+
+        # The seconds which we acquired must be (almost) equal to the ones
+        # specified in the CRON expression and they must be consecutive.
+        time1_second_found = None
```

Comment:
Let's base this not on physical time, but on the names of the tables. You can see in other tests that one can specify for the export timestamp to be part of the name. Then you can use these timestamps to check whether they are exactly equal to two consecutive values in the schedule.